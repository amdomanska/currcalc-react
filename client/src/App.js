import React, { Component } from 'react';
import './App.css';

import { Dropdown } from './components/dropdown'

class App extends Component {

  state = {
      response: ''
    };

    componentDidMount() {
      this.callApi()
        .then(res => this.setState({ response: res.express }))
        .catch(err => console.log(err));
    }

    callApi = async () => {
      const response = await fetch('/main');
      const body = await response.json();

      if (response.status !== 200) throw Error(body.message);

      return body;
    };

  render() {
    return (
      <div className="App">
        <div>
          <Dropdown />
        </div>

      </div>
    );
  }
}

export default App;
