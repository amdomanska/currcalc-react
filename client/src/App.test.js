import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import fetchMock from 'fetch-mock'

fetchMock.get('*', JSON.stringify({
  'rates': {
    'GBP': 0.5
  }
}));

it('renders without crashing', () => { /* eslint no-undef: "off" */
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
