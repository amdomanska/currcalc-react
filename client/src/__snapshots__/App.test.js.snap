// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders default state 1`] = `
<div
  className="App"
>
  <div>
    <p
      className="result-first-line hidden"
    >
       
      1
       
      Needs to be calculated!
       equals 
    </p>
    <p
      className="result-second-line hidden"
    >
       
       
      Needs to be calculated!
       
    </p>
  </div>
  <div
    className="rowC"
  >
    <div>
      <input
        className="value"
        onChange={[Function]}
        type="text"
        value={1}
      />
    </div>
    <div>
      loading...
    </div>
  </div>
  <div
    className="rowC"
  >
    <div>
      <input
        className="value"
        onChange={[Function]}
        type="text"
        value={1}
      />
    </div>
    <div>
      loading...
    </div>
  </div>
  <footer>
    <p
      className="footer"
    >
      @ data received from fixer.io
    </p>
  </footer>
</div>
`;
